src/hotspot/share/gc/shared/cardTable.hpp

```diff
@@ -57,7 +57,7 @@ class CardTable: public CHeapObj<mtGC> {
   static constexpr int max_covered_regions = 2;
 
   // The covered regions should be in address order.
-  MemRegion* _covered;
+  MemRegion _covered[max_covered_regions];
```

Comment:
The whole concept of covered_regions doesn't seem to belong in this class.  G1CardTable derives from
CardTable, but doesn't appear to use this at all.

Reply:
Yeah, it looks like there are a bunch of hierarchy / abstraction bugs in this area.  OK, I'll agree this is a
step-wise improvement, with hopefully more to come.